### DISCUSSION TITLE
Scheduled Flow for Full Recalc CMDT-driven Invocable" Apex action does not work, but manual Recalc Does

### DISCUSSION THREAD

**Discussion by tony-jones-tanium** [jamessimone/apex-rollup#389] [Q&A]
Hi Team, 

We have a Scheduled Flow built to run, and its calling a handful of Rollups that have been added to the Apex Action, by comma separated API Names. A few of the rollups work, but a couple are not working when triggered by the Flow. When using the manual recalc, they work just fine. 

Also, when I remove the API names of the Rollups, not working, and create a separate Scheduled Flow for just them, they work just fine. Is there a limit on how many Rollups can be added to the Full Recalc?

The flow is just scheduled to run and does nothing else besides call the Rollups. 

![Screen Shot 2022-12-01 at 10 21 31 AM](https://user-images.githubusercontent.com/103075625/205130515-8510c8ee-5536-4ab6-bee5-a4e433a48ff4.png)

**Comment by jamessimone** [comment]
@tony-jones-tanium quick question on this - are the API names that _aren't_ working in the first flow part of a separate SObject?

**Reply by tony-jones-tanium** [reply]
They are technically in both. What I have been doing to verify if they work, is I have all rollups in one flow (full recalc), and two are not firing. I deactivate that flow, and active the flow that just has the two rollups that weren't working, and they now work.

**Reply by tony-jones-tanium** [reply]
So the recalc of all of the Rollups looks like this: Jira_Issue_Count,Jira_Open_Issue_Count,Account_Tasks_Count,Business_Objectives_Count,TRA_Business_Driver,TRA_POC_Count

Then I verified the one that aren't working by creating a seperate flow with just these: 
Jira_Issue_Count,Jira_Open_Issue_Count

And then it works just fine. 

Also, the Parent Object is the same for all of these Rollups (Account).

**Reply by jamessimone** [reply]
@tony-jones-tanium OK that's interesting! I will take a look through and see what I can see. In the meantime, it might be helpful if you're able to generate a log (or see what logs get generated) when debugging the flow with `Is Rollup Logging Enabled` set to true on the Org Default Rollup Control. In specific, I'm interested to see what log messages get captured related to those two `Jira_` rollups.

My suspicion is that there's a deferral occurring that isn't properly handling those last two metrics - full recalcs process the rollups in reverse order, which is why it seems like those two at the beginning aren't being processed. However, that's not that many rollups to have configured on a parent by any means - six rollups - and that's where the logs may prove helpful as I log out exactly why a rollup in the midst of getting processed has to stop (due to whatever limit is getting exceeded). One possibility is that the heap size limit is being exceeded - I was a bit too cautious (in my opinion) in prior versions of Apex Rollup with what constituted the heap size limit, and I quite literally just released a new version of Apex Rollup that may address that particular issue.

Anyway - all of that is to say: let's see what info you can gather from the log messages that get generated, and I'll look through things on my end to see if there are any other likely culprits.

**Reply by tony-jones-tanium** [reply]
Thanks @jamessimone! Attached is a copy of the Log that didn't perform the Jira Rollups. 
[Uploading Rollup Log Messages 12.csv…]()

**Reply by jamessimone** [reply]
@tony-jones-tanium I'm not sure that attachment came through correctly - at the moment it's just a link to this discussion? Let me know if you see differently.

**Reply by tony-jones-tanium** [reply]
[Rollup Log Messages 12.csv](https://github.com/jamessimone/apex-rollup/files/10136153/Rollup.Log.Messages.12.csv)
See if this one worked!

**Reply by jamessimone** [reply]
There we go! It's possible that the log was truncated, however - I don't see any messages about timing out. That being said, I notice you're six versions behind on Apex Rollup, and there are definitely some efficiency improvements to the code path you're trying out that have since been implemented which may help.

I've also begun work on the _next_ version of Apex Rollup, `v1.5.36` based on some of my findings from this conversation which I expect will help to even further optimize things.

**Reply by jamessimone** [reply]
@tony-jones-tanium I've also just released #390 with the result of my findings from looking at your log file. I think that the combination of that and #387 will serve to close the loop on this one, but I'm looking forward to hearing more. Rest assured that if there are any other gaps, I'll work to close them for you as soon as possible.
